assets/app/features/upgrade/controllers/upgrade-nodes.controller.js

```diff
@@ -58,11 +70,40 @@
         function beginUpgradeNodes() {
             vm.nodesUpgrade.running = true;
 
-            upgradeFactory.upgradeNodes()
+            if (vm.nodesUpgrade.upgradedNodes === 0) {
```

Comment:
This check is not correct. E.g. if something fails after first node is upgraded user should be able to continue the upgrade (re-try). This code assumes that if some node was upgraded then upgrade was postponed.

Reply:
I've not encountered the error case that you mentioned. I don't know what the upgrade status call would return in that case. I can only code with the information I have. Will the steps.nodes.status return 'failed' while the upgraded_nodes return a value grater than 0? How about the 'current_substep_status' key? If you could provide the keys and values that I should check for, it would be helpful.

Reply:
One of my test clouds is currently in such state. This is 6-7 upgrade but the responses should be pretty similar. You can check this at: http://crowbar.vh10.cloud.suse.de/upgrade/#/upgrade/upgrade-nodes

Reply:
Thanks for letting me look at your system. It's very helpful to see the upgrade status. I'll update the code.